### PR TITLE
Adequacy patch: 2 rounds run / boundary update / output

### PR DESCRIPTION
**Pull request by hugo-antoine-rtei**

### CODE REVIEW

File: src/solver/simulation/economy.cpp [hunk 1]
```diff
@@ -146,7 +146,53 @@ bool Economy::year(Progression::Task& progression,
...
+                    for(int step = 0; step < numberOfTimesteps; ++step)
+                    {
+                        pProblemesHebdo[numSpace]->ResultatsHoraires[pays]->ValeursHorairesDENS[step]
+                            = pProblemesHebdo[numSpace]->ResultatsHoraires[pays]->ValeursHorairesDeDefaillancePositive[step];
+                        // logs.debug() << "##0:" << pays << ":" << step << ":" << pProblemesHebdo[numSpace]->ResultatsHoraires[pays]->ValeursHorairesDENS[step];
+                    }
+                }            
+
+                //todo need to change interconnection link property before second run
+                ::SIM_RenseignementProblemeHebdo(*pProblemesHebdo[numSpace], state, numSpace, hourInTheYear); //??? it is correct
```
Comment (Milos-RTEi) [inline src/solver/simulation/economy.cpp]: What does this line do ? 
The rest of the code is clear to me. 
I'll try to test it.

Comment (rtei-pw) [reply src/solver/simulation/economy.cpp]: reinit/reset something; we are asking rte; you can have a look if you find something

**Review by Milos-RTEi — APPROVED** [review]

**Review by rtei-pw — APPROVED** [review]

**Comment by rtei-pw** [comment]
@hugo-antoine-rtei you can review this PR and merge it when OK. We can leave the ::SIM_RenseignementProblemeHebdo problem there or wait for its resolution.

File: src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp [hunk 2]
```diff
@@ -394,21 +394,36 @@ void OPT_InitialiserLesBornesDesVariablesDuProblemeLineaire(PROBLEME_HEBDO* Prob
...
+                    {
+                        if(ProblemeHebdo->ResultatsHoraires[Pays]->ValeursHorairesDENS[PdtHebdo] >= 0.)
+                        {
+                            Xmax[Var] = ProblemeHebdo->ResultatsHoraires[Pays]->ValeursHorairesDENS[PdtHebdo] + 1e-5;
+                            // logs.debug() << "opt_gestion_bornes:" << Pays << ":" << PdtHebdo << ":Xmax[Var] = " << Xmax[Var];
+                        }
+                        else
+                            Xmax[Var] = 0.;
+                    }
+                }
```
Comment (Milos-RTEi) [inline src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp]: Creating test cases when realized.
From specifications: "additional constraint is introduced in the optimization problem for physical nodes that are declared in the adequacy patch domain (nodes declared as “2” in “adequacy_patch.ini” file". 
Am I missing something ? or I do not see if loop where ENS<= DENS **only if area type is 2**?

Comment (rtei-pw) [reply src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp]: Nice. Fixed

**Review by Milos-RTEi — APPROVED** [review]

File: src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp [hunk 3]
```diff
@@ -401,9 +401,19 @@ void OPT_InitialiserLesBornesDesVariablesDuProblemeLineaire(PROBLEME_HEBDO* Prob
...
                     Xmax[Var] = 0.;
 
+                // adq patch: update ENS <= DENS in 2nd run
+                if (ProblemeHebdo->UseAdequacyPatch == true
+                    && ProblemeHebdo->AdequacyFirstStep == false)
+                    if (ProblemeHebdo->AreaAdequacyPatchMode[Pays]
+                        == Data::adqmPhysicalAreaInsideAdqPatch)
+                        Xmax[Var]
+                          = ProblemeHebdo->ResultatsHoraires[Pays]->ValeursHorairesDENS[PdtHebdo]
+                            + 1e-5;
```
Comment (hugo-antoine-rtei) [inline src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp]: Why do we need that +1e-5 ?

Comment (rtei-pw) [reply src/solver/optimisation/opt_gestion_des_bornes_cas_lineaire.cpp]: removed

**Pull request merged**